pkg/local_object_storage/engine/delete.go

```diff
@@ -58,102 +55,25 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 		defer elapsed(e.metrics.AddDeleteDuration)()
 	}
 
-	var locked struct {
-		is  bool
-		err apistatus.ObjectLocked
-	}
-	var splitInfo *objectSDK.SplitInfo
-
-	// Removal of a big object is done in multiple stages:
-	// 1. Remove the parent object. If it is locked or already removed, return immediately.
-	// 2. Otherwise, search for all objects with a particular SplitID and delete them too.
-	e.iterateOverSortedShards(prm.addr, func(_ int, sh hashedShard) (stop bool) {
-		var existsPrm shard.ExistsPrm
-		existsPrm.SetAddress(prm.addr)
-
-		resExists, err := sh.Exists(existsPrm)
-		if err != nil {
-			if shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
-				return true
-			}
-
-			var splitErr *objectSDK.SplitInfoError
-			if !errors.As(err, &splitErr) {
-				if !shard.IsErrNotFound(err) {
-					e.reportShardError(sh, "could not check object existence", err)
-				}
-				return false
-			}
-			splitInfo = splitErr.SplitInfo()
-		} else if !resExists.Exists() {
-			return false
-		}
-
-		var shPrm shard.InhumePrm
-		shPrm.MarkAsGarbage(prm.addr)
-		if prm.forceRemoval {
-			shPrm.ForceRemoval()
-		}
-
-		_, err = sh.Inhume(shPrm)
+	if !prm.forceRemoval {
+		locked, err := e.IsLocked(prm.addr)
```

Comment:
```suggestion
		locked, err := e.isLocked(prm.addr)
```
no need to acquire `blockExec` mutex again